spec/models/status_spec.rb

```diff
@@ -7,4 +7,24 @@
       it { is_expected.to validate_length_of(:body).is_at_most(300) }
     end
   end
+
+  describe 'associations' do
+    context 'replies' do
+      it { is_expected.to have_many(:replies) }
+    end
+
+    context 'user' do
+      it { is_expected.to belong_to(:user) }
+    end
+
+    context 'media' do
+      it 'is expected to not allow more than 4 medium objects to be attached' do
+        status = Status.new(body: 'whatever')
+        5.times do
+          status.media << Medium.new(kind: 1, url: 'https://google.com/img.png')
+        end
+        is_expected.not_to be_valid
+      end
+    end
+  end
```

Comment:
This might cover your test case that asks if  a replied_to_status belongs to Status.

If you try to validade it without using "without_validating_presence" it will act as every Status require an original Status to be instantiated and eventually fail.

```suggestion
  describe 'associations' do
    context 'replies' do
      it { is_expected.to have_many(:replies) }
    end
    
    context 'replied_to_status' do
      it { is_expected.to belong_to(:replied_to_status).class_name('Status').without_validating_presence }
    end

    context 'user' do
      it { is_expected.to belong_to(:user) }
    end

    context 'media' do
      it 'is expected to not allow more than 4 medium objects to be attached' do
        status = Status.new(body: 'whatever')
        5.times do
          status.media << Medium.new(kind: 1, url: 'https://google.com/img.png')
        end
        is_expected.not_to be_valid
      end
    end
  end
```